pkg/oci/oci.go

```diff
@@ -537,11 +571,18 @@ func DeleteGadgetImage(ctx context.Context, image string) error {
 		return fmt.Errorf("resolving image: %w", err)
 	}
 
-	images, err := getGadgetImages(ctx, ociStore)
+	images, err := getGadgetImages(ctx, ociStore.Store)
 	if err != nil {
 		return fmt.Errorf("listing images: %w", err)
 	}
 
+	defer func() {
+		if err != nil {
+			return
+		}
+		err = ociStore.saveIndexWithLock()
```

Comment:
This function calls `ociStore.GC(ctx)` line 608.

But the documentation of `GC` says:
```go
// GC removes garbage from Store. Unsaved index will be lost. To prevent unexpected
// loss, call SaveIndex() before GC or set AutoSaveIndex to true.
```
With this PR, SaveIndex() is no longer called **before** GC().

Reply:
Good point. I've read this comment but I didn't get the idea before. I updated it to call `saveIndexWithLock` before GC(). I think GC() can still be racy at this point, but I'll explicitly ignore this for now.